modules.py

```diff
@@ -288,4 +304,80 @@ def rebin(coords, wavelength, size, detector_scale):
 
     x = (coords[1]-size//2.)*detector_scale/scale_factor+(size//2.)
     y = (coords[0]-size//2.)*detector_scale/scale_factor+(size//2.)
-    return y, x
\ No newline at end of file
+    return y, x
+
+
+class PolyPSF(OpticalArray):
+    """
+    Polychromatic PSF class. Inherits from OpticalArray.
+    """
+    def __init__(self, band, spectral_type='A', size=505, scale=0.01):
+        super(PolyPSF, self).__init__(size, poly=True, scale=scale)
+        self.band = band
+        self.spectral_type = spectral_type
+        self._wavelength_contributions = None
+        self.sed_file = 'sed_%s0_fe0.txt' % spectral_type.lower()
+        self._x = None
+        self._flux = None
+        self.name = 'polychromatic_psf'
+        self._dist = self.dist
+
+    def get_sed(self):
+        """
+        Reads the SED profile from files and get the spectral energy distribution.
+        :return: self._x, self._flux : two lists, wavelength and its associated flux.
+        """
+        if self._x is not None and self._flux is not None:
+            return self._x, self._flux
+        han = open('SED/%s' % self.sed_file)
+        data = han.readlines()
+        han.close()
+
+        x, flux = [], []
+        for i in range(len(data)):
+            xy = data[i].split()
+            if xy:
+                x.append(float(xy[0]))
+                flux.append(float(xy[1]))
+        self._x = x
+        self._flux = flux
+        return
+
+    def wavelength_contributions(self):
+        """
+        Computes the contributions of all 10 evenly-spaced wavelength in the filter band.
+        :return: wavelength_contributions. tuple of lists, containing the 10 wavelength and their relative contribution
```

Comment:
You say you are making a 'return' but you don't actually return anything?  This appears to be an initialization of the wavelength_contributions array?